### PR TITLE
Dep/rst complex

### PR DESCRIPTION
**Pull request by mts299**
Removed the RST defined `complex` struct and implemented the `complex.h` C defined library to address #350 

Test 
``` bash
make.build; make.code 
```
and run various commands to compare with the `develop` branch. Should be no difference.

### CODE REVIEW

**Comment by ecbland** [comment]
@mts299 I'm getting different values of `model.vel.median` from `map_addmodel` on this branch compared to `develop`. There are a lot of changes to `cnvmodel.c` so I'm wondering if something has crept in there?

**Comment by mts299** [comment]
Woops forgot a lot of that changed. Thank you for testing I will take a look tomorrow morning!

File: codebase/superdarn/src.lib/tk/cnvmodel.1.0/src/cnvmodel.c [hunk 1]
```diff
@@ -556,20 +563,20 @@ struct model *interp_coeffs(int ih, float tilt, float mag, float cang, int imod)
...
-                creal(Dp[k])*afn*mgn*dtp + creal(Ep[k])*afp*mgp*dtn +
-                creal(Fp[k])*afn*mgp*dtn + creal(Gp[k])*afp*mgn*dtn +
-                creal(Hp[k])*afn*mgn*dtn, cimag(Ap[k])*afp*mgp*dtp +
-                cimag(Bp[k])*afn*mgp*dtp + cimag(Cp[k])*afp*mgn*dtp +
-                cimag(Dp[k])*afn*mgn*dtp + cimag(Ep[k])*afp*mgp*dtn +
-                cimag(Fp[k])*afn*mgp*dtn + cimag(Gp[k])*afp*mgn*dtn +
-                cimag(Hp[k])*afn*mgn*dtn);
-
-      }
+      // TODO: I think this is duplicate code which makes more susceptable bugs
```
Comment (egthomas) [inline codebase/superdarn/src.lib/tk/cnvmodel.1.0/src/cnvmodel.c:566]: This is not duplicate code and should not be removed - `ptr->aoeff_n` and `ptr->aoeff_p` are both necessary.

**Comment by ecbland** [comment]
Ok, with that last commit I now get identical results for the convection mapping routines on this branch and `develop`.
I will double check the rest of this code ASAP but I think it is close to being done :+1:

**Comment by ecbland** [comment]
Ok, I think this one is good to merge. Thanks @mts299!

**Comment by mtwalach** [comment]
Sorry just saw this - I guess you don't need me to test it anymore then! 🙃

**Comment by ecbland** [comment]
> @mtwalach  Sorry just saw this - I guess you don't need me to test it anymore then! 🙃

Actually, could please check that you are able to compile RST on `develop` (including the `sim_*` libraries)? Then we will know if this PR solves your compilation problem on Catalina. My testing of this PR was only to check that no functionality was changed.

**Comment by mtwalach** [comment]
Okay, I'll test it next week!

**Pull request merged**